### DISCUSSION TITLE
getting cutoff from server when ipsec is up with install_routes = no

### DISCUSSION THREAD

**Discussion by irvx0** [strongswan/strongswan#1533] [Q&A]
Hi. 

I'm want to move from ipsec policy to ipsec route based tunnels. Need to build a ipsec tunnels (mesh) between all servers, ip  route based with BGB. But first i'm trying in lab to build correct configuration with baby steps.

I want to use 0.0.0.0/0 definition on left/right definition, so the servers can forward traffic or accept them. I will control routing via BGP and with iptables. 

When i'm using  in /etc/strongswan.conf
_install_routes = no
routing_table = 0_

and left/right in ipsec.conf as example
192.168.80.0/24

everything works fine. routes are not installed and i can access routing servers from NAT server (192.168.70.1). However if I will use 0.0.0.0/0 in left/right - routes are still not injected into routing table, but i'm lossing connectivity from 192.168.70.1 (nat server) to 192.168.70.200 (ipsec server). Basically ipsec server is cut of from network and can't event establish ipsec tunnels (as it is cut off). 
ip route looks normal and valid. 

I believe that this is due to the fact that if install_routes is disabled, _a more efficient lookup for source and next-hop addresses is used. Since version 5.5.2._ (based on documentation) How i can control that efficient lookup or disable it completely so i can install routes by BGP and ipsec tunnels can accept all networks that I want to push thru them.

os routers: debian 11
strongswan: Version: 5.9.1-1+deb11u3

_/etc/ipsec.conf
config setup
        strictcrlpolicy=no
        uniqueids = yes

conn r1****r2
        keyexchange=ikev2
        authby=secret
        auto=route
        type=tunnel
        left=192.168.70.1
        leftsubnet=0.0.0.0/0
        leftid=r1-router.2way.local
        right=192.168.71.1
        rightsubnet=0.0.0.0/0
        rightid=r2-router.2way.local
        ike=aes256-sha1-modp1024
        ikelifetime=28800s
        esp=aes256-sha1-modp1536
        compress=no
        fragmentation=yes
        forceencaps=yes
        dpdaction=restart
        dpddelay=30s
        rekey=no
        keyingtries=1_

_strongswan.conf 

charon {
        load_modular = yes
        plugins {
                include strongswan.d/charon/*.conf
        }
        routing_table = 0
        install_routes = no
}

include strongswan.d/*.conf_

ip address (on r1 router). R2 is same except +1 in 3 octet. i.e 192.168.71.1 / 192.168.81.1 etc.

_1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: ens192: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 00:50:56:ba:5f:f4 brd ff:ff:ff:ff:ff:ff
    altname enp11s0
    inet 192.168.70.1/24 brd 192.168.70.255 scope global ens192
       valid_lft forever preferred_lft forever
3: ens224: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 00:50:56:ba:b5:24 brd ff:ff:ff:ff:ff:ff
    altname enp19s0
    inet 192.168.80.200/24 brd 192.168.80.255 scope global ens224
       valid_lft forever preferred_lft forever_

ip route
_default via 192.168.70.200 dev ens192 onlink
192.168.70.0/24 dev ens192 proto kernel scope link src 192.168.70.1
192.168.80.0/24 dev ens224 proto kernel scope link src 192.168.80.200_

ipsec status (if left/right is 192.168.8x.0/24)
with 0/0 it is not established.

_Routed Connections:
    r1****r2{1}:  ROUTED, TUNNEL, reqid 1
    r1****r2{1}:   192.168.80.0/24 === 192.168.81.0/24
Security Associations (1 up, 0 connecting):
    r1****r2[1]: ESTABLISHED 2 hours ago, 192.168.70.1[r1-router.2way.local]...192.168.71.1[r2-router.2way.local]
    r1****r2{2}:  INSTALLED, TUNNEL, reqid 1, ESP in UDP SPIs: cbc0df40_i cdffa585_o
    r1****r2{2}:   192.168.80.0/24 === 192.168.81.0/24_

**Comment by tobiasbrunner** [comment]
> I believe that this is due to the fact that if install_routes is disabled, _a more efficient lookup for source and next-hop addresses is used._

No, that has nothing to do with it. It's simply that you are not actually using a route-based VPN (disabling `install_routes` is not what makes a VPN route-based) and that the IPsec policies will capture all traffic with `0.0.0.0/0` on both ends.

You have to either use marks, so only traffic marked via firewall or a VTI device matches the policies, or use XFRM interfaces that are associated with the policies and SAs (note that the latter can only be configured via swanctl.conf). Please see the docs for more information on [route-based VPNs](https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html).

**Reply by irvx0** [reply]
Thanks @tobiasbrunner for clarification. I was hoping that i can test element by element of that configuration. I will create then VTI and apply mark in config files. Will see where I will go from there. 

My understanding was that if I don't send traffic using route or iptables into ipseec tunnel, it will not capture traffic itself - therefore I just setup policy as 0/0 so tunnel will accept everything that I decide to push thru that, even if i'm not creating yet VTI. Thanks for clarification.
